packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -83,142 +88,69 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
-                    .unwrap()
-                    .to_rust_string_lossy(&mut try_catch);
+        let namespace: Object<'_> = {
+            let pkg = match ModuleMap::import_module(scope, &path) {
+                Ok(s) => s,
+                Err(Error::MissingDependency(_)) => {
+                    tracing::debug!(
+                        "Couldn't read package file. It might intentionally not exist."
+                    );
+                    // Packages don't have to use JS.
+                    let undefined = v8::undefined(scope).into();
+                    let fns = v8::Array::new_with_elements(scope, &[undefined; 3]);
 
-                Error::PackageImport(
-                    path.clone(),
-                    format!("Couldn't execute package setup: {exception_string}"),
-                )
-            })?;
+                    return Ok(JsPackage { fns });
+                }
+                Err(err) => return Err(err),
+            };
 
-            pkg.try_into().map_err(|err| {
-                Error::PackageImport(
-                    path.clone(),
-                    format!("Could not convert package setup return value into Function: {err}"),
-                )
-            })?
+            pkg.get_module_namespace()
+                .to_object(scope)
+                .expect("Module is not instantiated")
         };
 
-        let args = &[embedded.hash_util, embedded.hash_stdlib];
-        let fns: Array<'_> = {
-            let global_context = context.global(scope);
-            let fns = call_js_function(scope, pkg, global_context.into(), args).map_err(|err| {
-                Error::PackageImport(path.clone(), format!("Couldn't call package: {err}"))
-            })?;
-
-            fns.try_into().map_err(|err| {
-                Error::PackageImport(
-                    path.clone(),
-                    format!(
-                        "Couldn't convert package return value to array of package functions: \
-                         {err}"
-                    ),
-                )
-            })?
-        };
+        let fn_names = ["start_experiment", "start_sim", "run_task"];
+        let fns = fn_names
+            .into_iter()
+            .map(|fn_name| {
+                let mut try_catch = v8::TryCatch::new(scope);
+
+                let js_fn_name = new_js_string(&mut try_catch, fn_name);
+                // Get the function `Value` from the namespace
+                let func_or_undefined: Value<'_> = namespace
+                    .get(&mut try_catch, js_fn_name.into())
+                    .ok_or_else(|| {
+                        let exception = try_catch_exception(&mut try_catch).unwrap();
 
-        if fns.length() != 3 {
-            return Err(Error::PackageImport(
-                path.clone(),
-                "Unexpected amount of returned arguments".into(),
-            ));
-        }
+                        Error::PackageImport(
+                            name.to_string(),
```

Comment:
Previously this was the path, not the package name. Is there a specific reason to change this?

Reply:
I might have assumed it was the same... I don't remember
Fixed in https://github.com/hashintel/hash/pull/493/commits/0b92fb75c8fa12658eba320ee8f1cd34e365c0ff